packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -885,6 +885,10 @@ function resumeMountClassInstance(
     nextContext = getMaskedContext(workInProgress, nextLegacyUnmaskedContext);
   }
 
+  if (workInProgress.updateQueue === null) {
+    initializeUpdateQueue(workInProgress);
```

Comment:
This looks more like it's fixing symptoms.

The queue is emptied in `resetWorkInProgress` if `current === null`: 
https://github.com/facebook/react/blob/31a9e391f757d374ce0065da6735f08137151507/packages/react-reconciler/src/ReactFiber.js#L541-L549

In `updateClassComponent` we resume in `resumeMountClassInstance` if `current === null` which straight goes to `processUpdateQueue` assuming the `updateQueue` has been initialized (but it isn't anymore).

Note that not reseting the `updatedQueue` in `resetWorkInProgress` fixes this as well while not failing any other test.

Reply:
If the type were sound, the `updateQueue` on a class fiber should always be non-null. So my preference is to put it there instead.

But also it's classes so whatever :D

Reply:
If you do move the fix to `resetWorkInProgress`, I would add a dev-only check to `resumeMountClassInstance` to guard against a regression.

```js
if (__DEV__) {
  if (workInProgress.updateQueue === null) { 
    console.error('Internal error: Class fiber is missing an updateQueue');
  }
}
```

Reply:
- don't clear updateQueue in resetWorkInProgress
- add warning if updateQueue is empty in resumeWorkInProgress:
  > Class fiber is missing an updateQueue. This error is likely caused by a bug in React. Please file an issue.